spec/support/shared_examples/maintenance_commands.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "validates domain existence" do |command:|
+  context "when app has no domain" do
+    let!(:app) { dummy_test_app("nothing") }
+
+    it "raises error" do
+      result = run_cpflow_command(command, "-a", app)
+
+      expect(result[:status]).not_to eq(0)
+      expect(result[:stderr]).to include("Can't find domain")
+    end
+  end
+end
+
+RSpec.shared_examples "validates maintenance workload existence" do |command:|
+  context "when maintenance workload does not exist" do
+    let!(:app) { dummy_test_app("default", create_if_not_exists: true) }
+
+    it "raises error" do
+      result = run_cpflow_command(command, "-a", app)
+
+      expect(result[:status]).not_to eq(0)
+      expect(result[:stderr]).to include("Can't find workload 'maintenance'")
+    end
+  end
+end
+
+RSpec.shared_examples "switch maintenance mode command" do |action:|
```

Comment:
If we're following the same pattern as the ones above, this should be:

```suggestion
RSpec.shared_examples "switches maintenance mode command" do |action:|
```